### ISSUE TITLE
Equipping into the same NFT with 2 resources of same base

### ISSUE THREAD

**Issue by Swader** [rmrk-team/rmrk-substrate#110]
<img width="555" alt="Screenshot 2022-03-31 at 14 47 51" src="https://user-images.githubusercontent.com/1430603/161058154-4a9971a5-716f-4a5e-b645-bc418338c02a.png">

We should perhaps modify the Equip logic such that it pays attention to which resource is is equipping into as well.

**Comment by bmacer** [comment]
Currently we check only if there is *any* Resource for the Equipping NFT:
https://github.com/rmrk-team/rmrk-substrate/blob/92dc1563e3de2095d166043be8441353408c005f/pallets/rmrk-equip/src/functions.rs#L185-L196

I think we'll have to change the Equippings struct
https://github.com/rmrk-team/rmrk-substrate/blob/92dc1563e3de2095d166043be8441353408c005f/pallets/rmrk-equip/src/lib.rs#L82-L91

I believe we'd want to change it to something like this
```
	pub type Equippings<T: Config> = StorageNMap<
		_,
		(
			NMapKey<Blake2_128Concat, (CollectionId, NftId)>, // Equipper
			NMapKey<Blake2_128Concat, BaseId>, // Base ID
			NMapKey<Blake2_128Concat, SlotId>, // Slot ID
			BoundedResource<T>, // Equipper Resource ID
		),
		()
		OptionQuery,
	>;
```

This should allow us to define the Resource ID of the Base+Slot that we're equipping to.

Further, we can change the `equip` to allow for an Option<ResourceId>

Before
```
		pub fn equip(
			origin: OriginFor<T>,
			item: (CollectionId, NftId),
			equipper: (CollectionId, NftId),
			base: BaseId,
			slot: SlotId,
		) -> DispatchResult {
```

After

```
		pub fn equip(
			origin: OriginFor<T>,
			item: (CollectionId, NftId),
			equipper: (CollectionId, NftId),
			base: BaseId,
			slot: SlotId,
			resource: Option<ResourceId>, // <-- add this
		) -> DispatchResult {
```

In the event that resource is offered, we'll attempt to find that resource; if none is provided, we'll equip to the first resource we find, if any.

Maybe that will work?

**Comment by Ayuilos** [comment]
I think this could work, because RMRK2 standard on Kusama just didn't notice this situation.🫡
The standard only considers the Base level and does not consider the instance Resource level.

**Comment by bmacer** [comment]
I *think* this will be addressed with decoupling the equip extrinsic in #147
